modules/services/podman-linux/podman-lib.nix

```
@@ -83,8 +85,35 @@ in {
         else
           [ ];
 
+      checkImageTag = extraConfig:
+        let
+          buildSection = if (builtins.hasAttr "Build" extraConfig) then
+            extraConfig.Build
+          else
+            { };
+          imageTags = if (builtins.hasAttr "ImageTag" buildSection) then
+            buildSection.ImageTag
+          else
+            [ ];
+          containsRequiredTag = (if builtins.length imageTags > 0 then
+            builtins.elemAt imageTags 0
+          else
+            "") == "homemanager/${quadletName}";
+        in if (containsRequiredTag || builtins.length imageTags == 0) then
+          [ ]
+        else [{
+          assertion = false;
+          message = ''
+            In '${quadletName}' config. Build.ImageTag: '[ "${
+              builtins.concatStringsSep ''" "'' imageTags
+            }" ]' does not have 'homemanager/${quadletName}' at index 0.'';
+        }];
```

Does this tag really have to be in position 0? Perhaps something like

```suggestion
          containsRequiredTag = builtins.elem "homemanager/${quadletName}" imageTags;
          
          imageTagsStr = concatMapStringsSep " " toString imageTags;
        in [{
          assertion = imageTags == [ ] || containsRequiredTag;
          message = ''
            In '${quadletName}' config. Build.ImageTag: '[ ${imageTagsStr} ]' does not contain 'homemanager/${quadletName}'.'';
        }];
```

would work?

---

The reason that I had it forced to position 0 was so that you could easily discern which ones were from Home Manager and which ones were built outside of Home Manager. That said, it probably isn't really that important and this code is a lot more straight forward. Will make the swap.